src/detection/displayserver/displayserver_haiku.cpp

```diff
@@ -0,0 +1,73 @@
+extern "C" {
+#include "displayserver.h"
+#include "common/settings.h"
+#include "common/processing.h"
+}
+
+#include <math.h>
+
+#include <Application.h>
+#include <Screen.h>
+
+extern "C" void ffConnectDisplayServerImpl(FFDisplayServerResult* ds);
+
+static void detectDisplays(FFDisplayServerResult* ds)
+{
+    // We need a valid be_app to query the app_server here.
+    BApplication app("application/x-vnd.fastfetch-cli-fastfetch");
+    BScreen s; // default screen is the main one
```

Comment:
`BScreen s{};` or it looks uninitialized to me in a mainly-C project.

Reply:
It's an object, in C++ the default ctor is called, but OK.